src/entities/trade-aggregated.ts

```diff
@@ -0,0 +1,223 @@
+   */
+  public get executionPrice(): Price<TInput, TOutput> {
+    return (
+      this._executionPrice ??
+      (this._executionPrice = new Price(
+        this.inputAmount.currency,
+        this.outputAmount.currency,
+        this.inputAmount.quotient,
+        this.outputAmount.quotient
+      ))
+    )
+  }
+
+  /**
+   * The cached result of the price impact computation
+   * @private
+   */
+  private _priceImpact: Percent | undefined
+
+  /**
+   * Returns the percent difference between the route's mid price and the price impact
+   */
+  public get priceImpact(): Percent {
+    if (this._priceImpact) {
+      return this._priceImpact
+    }
+
+    let spotOutputAmount = CurrencyAmount.fromRawAmount(this.outputAmount.currency, 0)
+    for (const { route, inputAmount } of this.trades) {
+      const midPrice = route.midPrice
+      spotOutputAmount = spotOutputAmount.add(midPrice.quote(inputAmount))
+    }
+
+    const priceImpact = spotOutputAmount.subtract(this.outputAmount).divide(spotOutputAmount)
+    this._priceImpact = new Percent(priceImpact.numerator, priceImpact.denominator)
+
+    return this._priceImpact
+  }
+
+  public static async fromTrades<TInput extends Currency, TOutput extends Currency, TTradeType extends TradeType>(
+    trades: Trade<TInput, TOutput, TTradeType>[],
+    tradeType: TTradeType
+  ): Promise<AggregatedTrade<TInput, TOutput, TTradeType>> {
+    const sampleTrade = trades[0]
+    invariant(
+      trades.every(({ inputAmount: tradeInputAmount }) =>
+        sampleTrade.inputAmount.currency.equals(tradeInputAmount.currency)
+      ),
+      'INPUT_CURRENCY_MATCH'
+    )
+    invariant(
+      trades.every(({ outputAmount: tradeOutputAmount }) =>
+        sampleTrade.outputAmount.currency.equals(tradeOutputAmount.currency)
+      ),
+      'OUTPUT_CURRENCY_MATCH'
+    )
+
+    const totalInputFromTrades = trades
+      .map(({ inputAmount }) => inputAmount)
+      .reduce((total, cur) => total.add(cur), CurrencyAmount.fromRawAmount(sampleTrade.inputAmount.currency, 0))
+
+    const totalOutputFromTrades = trades
+      .map(({ outputAmount }) => outputAmount)
+      .reduce((total, cur) => total.add(cur), CurrencyAmount.fromRawAmount(sampleTrade.outputAmount.currency, 0))
+
+    return new AggregatedTrade({
+      trades,
+      inputAmount: totalInputFromTrades,
+      outputAmount: totalOutputFromTrades,
+      tradeType: tradeType
+    })
+  }
+
+  /**
+   * Construct a trade by passing in the pre-computed property values
+   * @param route The route through which the trade occurs
+   * @param inputAmount The amount of input paid in the trade
+   * @param outputAmount The amount of output received in the trade
+   * @param tradeType The type of trade, exact input or exact output
+   */
+  constructor({
+    trades,
+    inputAmount,
+    outputAmount,
+    tradeType
+  }: {
+    trades: Trade<TInput, TOutput, TTradeType>[]
+    inputAmount: CurrencyAmount<TInput>
```

Comment:
does not need to be specified, can just be a lazy getter that computes it from the trades